xmtp_proto/src/convert.rs

```diff
@@ -47,6 +52,67 @@ impl TryFrom<UploadKeyPackageRequest> for ClientEnvelope {
     }
 }
 
+impl TryFrom<OriginatorEnvelope> for KeyPackage {
+    type Error = crate::ProtoError;
```

Comment:
Probably more apt to make this a [conversion error](https://github.com/xmtp/libxmtp/blob/9f0dd2cfcfa484f95e19fd8b3ec7c3217567ba84/xmtp_proto/src/error.rs#L213) which is also defined in proto, but allows more information about lost types in the error